app/src/main/java/com/example/footballmanager/ui/screens/login/Login.kt

```diff
@@ -0,0 +1,174 @@
+    ) {
+        if (user == null) {
+            LogInOutButton(
+                onClickFunction = {
+                    val gso =
+                        GoogleSignInOptions
+                            .Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                            .requestIdToken(token)
+                            .requestEmail()
+                            .build()
+                    val googleSignInClient = GoogleSignIn
+                        .getClient(context, gso)
+                    launcher
+                        .launch(googleSignInClient.signInIntent)
+                },
+                text = stringResource(R.string.sign_in_via_google)
+            )
+        } else {
+            LogInOutButton(
+                onClickFunction = {
+                    Firebase.auth.signOut()
+                    user = null
+                },
+                text = stringResource(R.string.log_out)
+            )
+        }
+    }
+}
+
+@Composable
+fun LogInOutButton(onClickFunction: () -> Unit, text: String) {
+    ElevatedButton(
+        onClick = {
+            onClickFunction.invoke()
+        },
+        shape = RoundedCornerShape(dimensionResource(id = R.dimen.login_activity_button_rounded)),
+        modifier = Modifier
+            .height(dimensionResource(id = R.dimen.login_activity_button_height))
+            .fillMaxSize()
+            .padding(dimensionResource(id = R.dimen.login_activity_button_padding)),
+        colors = ButtonDefaults.buttonColors(
+            containerColor = Color.White,
+            contentColor = Color.Black
+        )
+    ) {
+        Image(
+            painter = painterResource(id = R.drawable.google_logo),
+            contentDescription = stringResource(R.string.google_logo),
+            modifier = Modifier.size(dimensionResource(id = R.dimen.login_activity_button_image_size)),
+            contentScale = ContentScale.Fit
+        )
+
+        Spacer(modifier = Modifier.width(dimensionResource(id = R.dimen.login_activity_button_spacer)))
+
+        Text(
+            text,
+            fontFamily = FontFamily.SansSerif,
+            fontWeight = FontWeight.ExtraBold,
+            fontSize = dimensionResource(id = R.dimen.login_activity_button_font_size).value.sp,
+            letterSpacing = dimensionResource(id = R.dimen.login_activity_button_font_spacing).value.sp
+        )
+    }
+}
+
+@Composable
+fun rememberFirebaseAuthLauncher(
+    onAuthComplete: (AuthResult) -> Unit,
+    onAuthError: (ApiException) -> Unit
+): ManagedActivityResultLauncher<Intent, ActivityResult> {
+    val scope = rememberCoroutineScope()
+    return rememberLauncherForActivityResult(
+        ActivityResultContracts
+            .StartActivityForResult()
+    ) { result ->
+        val task = GoogleSignIn
+            .getSignedInAccountFromIntent(result.data)
+        try {
+            val account = task
+                .getResult(ApiException::class.java)!!
```

Comment:
usually it is not safe to use the not-null assertion operator `!!`, it throws an exception if the value is null - consider using safe call `?.`